src/averell/core.py

```diff
@@ -1,29 +1,57 @@
 import importlib
+import logging
+import os
 from pathlib import Path
 
-import yaml
-
-from averell.utils import download_corpora
-from averell.utils import filter_features
+from .utils import CORPORA_SOURCES
+from .utils import download_corpora
+from .utils import filter_features
+from .utils import write_json
 
 DEFAULT_OUTPUT_FOLDER = Path.cwd() / "corpora"
 
-CORPORA_SOURCES = yaml.load("corpora.yaml", Loader=yaml.FullLoader)
-
 
 def get_corpora(corpus_indices=None, granularity=None,
                 output_folder=DEFAULT_OUTPUT_FOLDER):
     """
 
+    :param corpus_indices:
+    :param granularity:
+    :param output_folder:
+    :return:
     """
-    download_corpora(corpus_indices, output_folder)
     corpora_features = []
-    for index in corpus_indices:
-        folder_name = CORPORA_SOURCES[index]["properties"]['folder_name']
-        get_features = getattr(importlib.import_module(
-            CORPORA_SOURCES[index]["properties"]["reader"]), "get_features")
-        features = get_features(Path(output_folder) / folder_name)
-        if granularity is not None:
-            features = filter_features(features, index, granularity)
-        corpora_features.append(features)
-    return corpora_features
+    try:
+        download_corpora(corpus_indices, output_folder)
+        for index in corpus_indices:
+            folder_name = CORPORA_SOURCES[index]["properties"]['folder_name']
+            get_features = getattr(importlib.import_module(
+                CORPORA_SOURCES[index]["properties"]["reader"]), "get_features")
+            features = get_features(Path(output_folder) / folder_name)
+            for poem in features:
+                author_path = Path(output_folder) / folder_name / "averell" / "parser" / poem["author"].replace(" ", "")
+                if not author_path.exists():
+                    os.makedirs(author_path)
+                write_json(poem, str(author_path / poem["poem_title"].title().replace(" ", "")))
+            if granularity is not None:
+                granularities_list = CORPORA_SOURCES[index]["properties"][
+                    "granularity"]
+                if granularity in granularities_list:
+                    for poem in features:
+                        filtered_features = filter_features(poem, index, granularity)
+                        granularity_path = Path(
+                            output_folder) / folder_name / "averell" / granularity / poem["author"].replace(" ", "")
+                        if filtered_features:
+                            if not granularity_path.exists():
+                                os.makedirs(granularity_path)
+                            write_json(filtered_features, str(granularity_path / poem["poem_title"].title().replace(" ", "")))
```

Comment:
Line too long, can you fix this throughout the code please?

Reply:
Done